.github/workflows/testing.yml

```diff
@@ -8,6 +8,78 @@ on:
     branches: [ "main" ]
 
 jobs:
+    docker:
+        # This job will create a docker instance, so that it can be used when jest is run
+        name: create docker 
+        runs-on: ubuntu-latest
+        timeout-minutes: 1  # can be changed later
+        continue-on-error: true
+        permissions:
+            contents: read
+            security-events: write
+            actions: read # only required for a private repository by github/codeql-action/upload-sarif to get the Action run status
+        steps:
+          - name: Checkout code
+            uses: actions/checkout@v3
+
+          - name: Create env
+            working-directory: ./server
+            run: |
+              touch .env
+              echo NODE_ENV=development>> .env
+              echo PORT=3000>> .env
+              echo DB_HOST=mongodb>> .env
+              echo DB_NAME=roadwatch>> .env
+              echo DB_USER=root>> .env
+              echo DB_PASSWORD=password>> .env
+              echo DB_PORT=27017>> .env
+              echo JWT_SECRET=secret>> .env
+              echo JWT_EXPIRES_IN=24hr>> .env
+              cat .env
+
+          - name: Set up Docker instance
+            working-directory: ./server
+            run: docker-compose up
+
+    server_start:
+      name: start server
+      runs-on: ubuntu-latest
+      timeout-minutes: 1  # can be changed later
+      continue-on-error: true
+      permissions:
+          contents: read
+          security-events: write
+          actions: read # only required for a private repository by github/codeql-action/upload-sarif to get the Action run status
+      steps:
+        - name: Checkout code
+          uses: actions/checkout@v3
+
+        - name: Create env
+          working-directory: ./server
+          run: |
+            touch .env
+            echo NODE_ENV=development>> .env
+            echo PORT=3000>> .env
+            echo DB_HOST=mongodb>> .env
+            echo DB_NAME=roadwatch>> .env
+            echo DB_USER=root>> .env
+            echo DB_PASSWORD=password>> .env
+            echo DB_PORT=27017>> .env
+            echo JWT_SECRET=secret>> .env
+            echo JWT_EXPIRES_IN=24hr>> .env
+            echo NOTIFICATION_PUBLIC_KEY=BIQ8FWqgNZtyxHeCALbruZO8ttzxza2xT4BeNM2E4Ix9HYeAVwVDRkHWdzNqJTdhDj21uryLKRKJpOBhqnhUEr0>> .env
```

Comment:
uh no, if we need these kind of environment variables into this then we are going to have to figure out a different method or provide it a dummy one so that it can compile. These keys are kind of whatever, but imagine if this was sendgrid api or google maps api, you would've just exposed it, so I would prefer to find another way if possible.

I think these notif keys are needed in order for the server/client to start up properly, see if you can just provide some dummy one like "notifPubKey". Otherwise I feel like it a malicious user would be able to use these keys to send people subscribed notifications from our app

Reply:
I agree with Josh. Exposing both the public and private keys here would allow anyone to send notifications to anyone. For the purposes of testing, maybe generate a new set using the instructions I left in the README, or just create some dummy values.

Reply:
sorry about that, that was an oversight on my part. i have removed the actual key and just replaced it with DUMMY, and that seems to be working fine, since i'm not testing the push notifications in this PR. the testing seems to be working fine after this fix